examples/rust/digital_signature/src/lib.rs

```diff
@@ -0,0 +1,66 @@
+// Copyright 2022 Risc0, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use digital_signature_proof::{Message, Passphrase, SignMessageCommit, SigningRequest};
+use r0vm_host::{Proof, Prover, Result};
+use r0vm_serde::{from_slice, to_vec};
+
+pub struct SigningCompleteMessage {
+    proof: Proof,
+}
+
+impl SigningCompleteMessage {
+    pub fn get_commit(&self) -> Result<SignMessageCommit> {
```

Comment:
Maybe a separate get_identity() and get_message().  Also, maybe a plain 'verify()'?  I do think verify_and_get_commit() is like the all in one case, and the others are like ease of use, not super picky on this, just a thought.

Reply:
Updated in latest commit with the following public functions:

```rust
pub fn get_commit(&self) -> Result<SignMessageCommit>;
pub fn get_identity(&self) -> Result<r0vm_core::Digest>;
pub fn get_message(&self) -> Result<Message>;
pub fn verify(&self) -> Result<SignMessageCommit>;
```